### ISSUE TITLE
Discrepancy in Loss Function Implementation

### ISSUE THREAD

**Issue by yueyuwen-586** [zhuyu-cs/MeanFlow#27]
The loss function implementation in [loss.py] does not match the paper's adaptive loss description. The code uses (loss_mid^2 + 1e-3)^p for weights and loss_mid^2 for the loss, but the paper specifies loss_mid + 1e-3 and loss_mid. This results in a different loss than intended. Suggested fix: change weights = 1.0 / (loss_mid.detach() ** 2 + 1e-3).pow(self.adaptive_p) to weights = 1.0 / (loss_mid.detach() + 1e-3).pow(self.adaptive_p) and loss = weights * loss_mid ** 2 to loss = weights * loss_mid. Can you confirm if this was intentional or apply the fix? Thanks!
